File: src/app/src/main/java/com/example/screenservicetvapp/fragments/TextADInformationFragment.java
package com.example.screenservicetvapp.fragments;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.screenservicetvapp.ObjectExtensions;
import com.example.screenservicetvapp.R;

public class TextADInformationFragment extends Fragment {

    private String backgroundColor;
    private String description;
    private String name;
    private String textColor;
    private String textFont;

    private TextView textEditorTextView;

    public TextADInformationFragment() {
        // Required empty public constructor
    }

    public static TextADInformationFragment newInstance(String param1, String param2) {
        TextADInformationFragment fragment = new TextADInformationFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle bundle = getArguments();
        if (bundle != null) {
            description = bundle.getString("description");
            backgroundColor = bundle.getString("backgroundColor");
            name = bundle.getString("name");
            textColor = bundle.getString("textColor");
            textFont = bundle.getString("textFont");
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_text_ad_information, container, false);

        if(ObjectExtensions.isNullOrEmpty(description)) description = "Error: No text found in the data, republish.";
        textEditorTextView = view.findViewById(R.id.text_ad_information_textview);
        textEditorTextView.setText(Html.fromHtml(description));
        return view;
    }
}

File: src/app/src/main/java/com/onscreensync/tvapp/fragments/TextADInformationFragment.kt
package com.onscreensync.tvapp.fragments

import android.os.Bundle
import android.text.Html
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.onscreensync.tvapp.R
import com.onscreensync.tvapp.databinding.FragmentTextAdInformationBinding
import com.onscreensync.tvapp.utils.UiHelper
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class TextADInformationFragment : Fragment() {

    private var _binding: FragmentTextAdInformationBinding? = null
    private val binding get() = _binding!!

    private var backgroundColor: String? = null
    private var description: String? = null
    private var name: String? = null
    private var textColor: String? = null
    private var textFont: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let { bundle ->
            description = bundle.getString("description")
            backgroundColor = bundle.getString("backgroundColor")
            name = bundle.getString("name")
            textColor = bundle.getString("textColor")
            textFont = bundle.getString("textFont")
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentTextAdInformationBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val displayDescription = if (description.isNullOrEmpty()) {
            "Error: No text found in the data, republish."
        } else {
            description!!
        }

        binding.textAdInformationTextview.apply {
            text = Html.fromHtml(displayDescription)
            UiHelper.setTextViewFont(this, textFont)
            UiHelper.setTextViewColor(this, textColor)
        }

        binding.root.setBackgroundColor(UiHelper.parseColor(backgroundColor))
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        @JvmStatic
        fun newInstance(param1: String?, param2: String?) = TextADInformationFragment()
    }
}